src/StreamFactoryInterface.php

```diff
@@ -9,15 +9,11 @@
     /**
      * Create a new stream from a resource or a string.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
+     * String content will be used as the body of the stream. Files paths are not supported.
```

Comment:
>> a library should not depend on a implementation (utility package), it should rely on interfaces (PSR-17)

> For the functionality covered by those interfaces, of course. No one would disagree with that. What we disagree about is the scope.

Okey, I thought you were making this argument. Good. 

> IMO, the scope of an interoperability layer must be kept low-level - if you introduce higher-order abstractions that aren't strictly necessary for interoperability, you're merely burdening every developer with the same implementations, tests, etc. while gaining zero in terms of functionality.

Naturally. =)

> The only reason to go beyond the scope required for intereoperability, is to displace or suppress design diversity.

> I hope we're not trying to do that?

Of course not. =)

>> If we can reduce interoperability concerns to a single, consistent interface, and this enables someone to write a utility library that provides the same functionality and utility for all implementations of that interface, isn't that the ideal?

You do also say that you want to "design the interfaces based on what's minimally required to level the differences between libraries". 

When I read this paragraphs of yours it seams like you are suggesting to have a small interface in the PSR (ie only support `resource`) and then let someone create a utility package to make the PSR interface easy to use (ie including functions `createFromString`, `createFromPath`).

I may have misunderstood your argument and this might not be what you suggest. But the scenario I describe above is what I consider as a failure for this (and any) PSR. 

--------

I want to remind everybody about the goals of this organisation (in no particular order): 

* Figure out the best possible solution/interface
* Adapting to current leading implementations
* Be developer friendly

I know that most of you have knowledge and experience far beyond mine, but I do not think many of you have been implementing factory interfaces for PSR-7 as much as I have. I've been working with HTTPlug for a year now. I've been making loads of PRs to big and small libraries to implement factory interfaces like this. The argument I want to make is that we should be careful not to overlook the developer experience and ease of use. 

In my experience, people have a tendency to feel like an abstraction is unnecessary. If on top of that, the abstraction is not easy to use, I fear that it would be a high resistance for this PSR. If we have to inject 2-3 new objects in the constructor to create a stream people would say "Screw PSR-17, Lets just use `\GuzzleHttp\Psr7\stream_for()`". It does not matter how technically correct we are or how much I think they are wrong. 

Bottom line is, we *may* have to make sacrifices of being technically correct and having a perfect solution in order to be accepted by the community.